### PR TITLE
DEV: Adding in SSVD implementation

### PR DESCRIPTION
**Pull request by mortonjt**
My bad about the previous PRs - need to clean up the commit history.

Here's an implementation of SSVD and some unittests.

From my tests, it seems that this is actually quite darn accurate. Not sure what the error bounds are atm. May add them in a bit.

Btw. I added these methods to a new file `skbio/linalg`. May want to put this somewhere else, like in the `utils` folder

The main point is, we need to have these decompositions methods be global methods, otherwise there will surely be unnecessary API revamping in the future.

### CODE REVIEW

File: skbio/tests/test_linalg.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+class SSVDTests(TestCase):
+
+    def setUp(self):
+        np.random.seed(0)
+        self.mat1 = np.random.randn(9, 6)
+        self.mat2 = np.dot(self.mat1.T, self.mat1)
+
+    def test_ssvd(self):
+        np.random.seed(0)
```
Comment (wasade) [inline skbio/tests/test_linalg.py:29]: can do this in setup

File: skbio/tests/test_linalg.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        test_s, test_U, test_V = ssvd(mat1, k=3)
```
Comment (wasade) [inline skbio/tests/test_linalg.py]: `mat1` is undefined

File: skbio/tests/test_linalg.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        actual_U, actual_s, actual_V = scipy.sparse.linalg.svds(mat1, k=3)
+        npt.assert_allclose(-1*test_U, actual_U, rtol=4)
+        npt.assert_allclose(test_s, actual_s, rtol=4)
+        npt.assert_allclose(-1*test_V, actual_V, rtol=4)
+
+    def test_ssvd_eig(self):
+        np.random.seed(0)
+        test_s, test_U = ssvd(mat2, k=5, compute_v=False)
```
Comment (wasade) [inline skbio/tests/test_linalg.py]: `mat2` is undefined

Comment (mortonjt) [inline skbio/tests/test_linalg.py]: That was dumb of me.  I had trouble getting the test suite to work due to
errors in _principal_coordinate_analysis.py

Thanks for catching this!  I'll merge it as soon as we finalize where these
decomposition methods should go.

**Pull request merged**